src/rebuild/srv.c

```diff
@@ -618,6 +601,12 @@ rebuild_leader_status_check(struct ds_pool *pool, uint32_t map_ver,
 			}
 
 			D_FREE(targets);
+
+			if (is_rebuild_global_scan_done(rgt))
+				rgt->rgt_scan_done = 1;
+
+			if (is_rebuild_global_pull_done(rgt))
+				rgt->rgt_done = 1;
```

Comment:
The failed targets are VOS targets, but the scan/pull bits are per node, so if any VOS target in a node failed, the whole node will be marked as scan/pull done?

And what confused me is that: This old check down status then mark done logic didn't work without this change, then how can we pass existing rebuild test? The failed/killed node is already excluded from the scan/pull bitmap?

Reply:
oh, you are right, I should check the node status instead of target status.